src/renderer/components/+namespaces/namespace-select-filter.test.tsx

```diff
@@ -0,0 +1,252 @@
+      });
+
+      it("closes menu", () => {
+        expect(result.baseElement.querySelector("#react-select-namespace-select-filter-listbox")).toBeNull();
+      });
+
+      describe("when clicked again", () => {
+        beforeEach(() => {
+          result.getByTestId("namespace-select-filter").click();
+        });
+
+        it("renders", () => {
+          expect(result.baseElement).toMatchSnapshot();
+        });
+
+        it("shows 'test-2' as selected", () => {
+          expect(result.queryByTestId("namespace-select-filter-option-test-2-selected")).not.toBeNull();
+        });
+
+        it("does not show 'test-1' as selected", () => {
+          expect(result.queryByTestId("namespace-select-filter-option-test-1-selected")).toBeNull();
+        });
+
+        describe("when 'test-1' is clicked", () => {
+          beforeEach(() => {
+            result.getByText("test-1").click();
+          });
+
+          it("renders", () => {
+            expect(result.baseElement).toMatchSnapshot();
+          });
+
+          it("has only 'test-1' is selected in the store", () => {
+            expect(namespaceStore.contextNamespaces).toEqual(["test-1"]);
+          });
+
+          it("closes menu", () => {
+            expect(result.baseElement.querySelector("#react-select-namespace-select-filter-listbox")).toBeNull();
+          });
+
+          describe("when clicked again, then holding down multi select key", () => {
+            beforeEach(() => {
+              const filter = result.getByTestId("namespace-select-filter");
+
+              filter.click();
+              fireEvent.keyDown(filter, { key: "Meta" });
+            });
+
+            describe("when 'test-3' is clicked", () => {
+              beforeEach(() => {
+                result.getByText("test-3").click();
+              });
+
+              it("renders", () => {
+                expect(result.baseElement).toMatchSnapshot();
+              });
+
+              it("has both 'test-1' and 'test-3' as selected in the store", () => {
+                expect(new Set(namespaceStore.contextNamespaces)).toEqual(new Set(["test-1", "test-3"]));
+              });
+
+              it("keeps menu open", () => {
+                expect(result.baseElement.querySelector("#react-select-namespace-select-filter-listbox")).not.toBeNull();
+              });
+
+              it("does not show 'kube-system' as selected", () => {
+                expect(result.queryByTestId("namespace-select-filter-option-kube-system-selected")).toBeNull();
+              });
+
+              describe("when 'test-13' is clicked", () => {
+                beforeEach(() => {
+                  result.getByText("test-13").click();
+                });
+
+                it("has all of 'test-1', 'test-3', and 'test-13' selected in the store", () => {
+                  expect(new Set(namespaceStore.contextNamespaces)).toEqual(new Set(["test-1", "test-3", "test-13"]));
+                });
```

Comment:
Could also test that when one of them is clicked again it is no longer in  `namespaceStore.contextNamespaces`?